### DISCUSSION TITLE
Grouping with Row Selection is broken

### DISCUSSION THREAD

**Discussion by ghiscoding** [6pac/SlickGrid#931] [General]
@6pac I recently noticed that  [example-grouping-checkbox-row-select.html](http://6pac.github.io/SlickGrid/examples/example-grouping-checkbox-row-select.html) is broken but we don't have Cypress E2E tests for it so we never caught it. I assume it probably broke when we either removed jQuery or when we migrated to TypeScript. I'm not using this feature and you are the person who created it, so I'll let you decide if you want to look at it. Cheers

**Comment by 6pac** [comment]
This appears to be because the line in ```slick.groupitemmetadataprovider.ts``` method ```handleGridClick```:

        (item.selectChecked ? this._options.checkboxSelectPlugin.selectRows : this._options.checkboxSelectPlugin.deSelectRows)(rowIndexes);

is separating the function ```selectRows``` or ```selectRows``` from its parent object when calling, and hence ```this``` in the called method is undefined. This is causing row selection to fail.  
Replace the above line with

    if (item.selectChecked) { 
      this._options.checkboxSelectPlugin.selectRows(rowIndexes); 
    } else {
      this._options.checkboxSelectPlugin.deSelectRows(rowIndexes);
    }

**Reply by ghiscoding** [reply]
oh ok I see, you can go ahead with a PR then, we don't have Cypress tests for that Example which is why we didn't catch the error earlier
